tests/test_ssa_value.py

```diff
@@ -62,9 +63,31 @@ def test_invalid_ssa_vals(name: str):
         val.name_hint = name
 
 
-def test_owner():
+def test_rewrite_type():
+    """We can rewrite the type of a test SSA value."""
     val = TestSSAValue(i32)
+    rewriter = Rewriter()
+    new_val = rewriter.replace_value_with_new_type(val, i64)
+    assert val.type == i32
+    assert new_val.type == i64
+
+
+def test_replace_type():
+    class InvalidValue(SSAValue):
+        @property
+        def owner(self) -> Operation | Block:
+            """
+            An SSA variable is either an operation result, or a basic block argument.
+            This property returns the Operation or Block that currently defines a specific value.
+            """
+            ...
+
+    val = InvalidValue(i32)
+
+    rewriter = Rewriter()
+
     with pytest.raises(
-        ValueError, match="Attempting to get the owner of a `TestSSAValue`"
+        ValueError,
+        match="Expected OpResult or BlockArgument, got InvalidValue",
     ):
-        val.owner
+        rewriter.replace_value_with_new_type(val, i32)
```

Comment:
I don't think this test is relevant.
Or at least, we have so much piece of infrastructure that relies on the fact that a value is an operand or a block argument, that we should test these then (but I don't believe it's relevant).

Reply:
I don't understand your suggestion, should we keep it or remove it?